tsdb/series_file.go

```diff
@@ -445,8 +448,11 @@ func CompareSeriesKeys(a, b []byte) int {
 	}
 
 	// Read total size.
-	_, a = ReadSeriesKeyLen(a)
-	_, b = ReadSeriesKeyLen(b)
+	szA, a := ReadSeriesKeyLen(a)
+	szB, b := ReadSeriesKeyLen(b)
+	if szA == 0 && szB == 0 {
+		return 0
+	}
```

Comment:
Could this be a more complete test?  Something like this?  (Caveat, this may not be correct logic, 
 but I want to avoid the calls to `ReadSeriesKeyMeasurement` below for a zero length key)
```
    switch cmp := szA - szB; cmp {
    case szA != 0 && szB != 0:
        break
    case cmp > 0:
        return 1
    case cmp < 0:
        return -1
    case cmp == 0:
        return 0
    default:
    }
    // Read names.
    name0, a := ReadSeriesKeyMeasurement(a)
    name1, b := ReadSeriesKeyMeasurement(b)
```

Reply:
This makes sense 👍 

I was wondering too shouldn't this perhaps be consolidated into the check above?

https://github.com/influxdata/influxdb/blob/2556e3875f2f836e15f96c579292a58015afd8eb/tsdb/series_file.go#L440-L448

I'm failing the see the difference between the initial `if len(a) == 0 ...` portion which allegedly handles `nil` keys in a similar manner to what we're trying here, but then the call into `ReadSeriesKeyLen` below which would provide a `szA = 0` _should_ have already been skipped over from the prior check and early return.